tests/test_loop_fusion.py

```diff
@@ -0,0 +1,164 @@
+    for i in range(3):
+        x: int = 1
+        pk.printf("%d\n", x)
+        for k in range(3):
+            a : int = 2
+            pk.printf("%d\n", a)
+
+    for i in range(3):
+        x: int = 3
+        pk.printf("%d\n", x)
+        for k in range(3):
+            a : int = 4
+            pk.printf("%d\n", a)
+
+@pk.workunit
+def nested_triples(tid, v): 
+
+    pk.printf("%f\n", 0.5)
+    for i in range(3):
+        x: int = 1
+        pk.printf("%d\n", x)
+        
+        for k in range(3):
+            a : int = 2
+            pk.printf("%d\n", a)
+   
+    pk.printf("%f\n", 1.5)
+    for j in range(3):
+        y: int = 3
+        pk.printf("%d\n", y)
+
+        for k in range(3):
+            a : int = 4
+            pk.printf("%d\n", a)
+
+    pk.printf("%f\n", 2.5)
+    for j in range(3):
+        y: int = 5
+        pk.printf("%d\n", y)
+
+        for k in range(3):
+            a : int = 6
+            pk.printf("%d\n", a)
+
+@pk.workunit
+def nested_triples_noprint(tid, v): # removing prints in between loops should allow fusion?
+
+    for i in range(3):
+        x: int = 1
+        pk.printf("%d\n", x)
+        
+        for k in range(3):
+            a : int = 2
+            pk.printf("%d ", a)
+   
+
+    for j in range(3):
+        y: int = 3
+        pk.printf("%d\n", y)
+
+        for k in range(3):
+            a : int = 4
+            pk.printf("%d ", a)
+
+
+    for j in range(3):
+        y: int = 5
+        pk.printf("%d\n", y)
+
+        for k in range(3):
+            a : int = 6
+            pk.printf("%d ", a)
+
+@pk.workunit
+def view_manip_inbetween(tid, v): # I guess manually inspect c++? 
+    for i in range(3):
+        x: int = 1
+        pk.printf("%d\n", x)
+    
+    v[tid] = -1;
+    v[tid] = v[tid] + 3;
+
+    for j in range(3):
+        y: int = 2
+        pk.printf("%d\n", y)
+
+@pk.workunit
+def inner_scopes(tid, v): # 1 2 3 4 4 4 5 5 5 ...
+
+    for i in range(3):
+        x: int = 1
+        pk.printf("%d\n", x)
+
+        for j in range(3):
+            y: int = 2
+            pk.printf("%d\n", y)
+
+        for j in range(3):
+            y: int = 3
+            pk.printf("%d\n", y)
+
+            for j in range(3):
+                y: int = 4
+                pk.printf("%d\n", y)
+
+            for j in range(3):
+                y: int = 5
+                pk.printf("%d\n", y)
+
+@pk.workunit
+def nadir_fusable(tid, v):
+    for i in range(1):
+        x: int = 3
+        pk.printf("%d\n", x)
+        for j in range(x):
+            q: int = 0
+            pk.printf("lol %d\n", x)
+            for n in range(x, q):
+                pk.printf("%d\n", q)
+
+        for k in range(2):
+            pk.printf("print 1 %d\n", k)
+        up_here: int = 0
+
+    for j in range(1):
+        y: int = 4
+        x: int = y
+        # z: float = v[0]
+        # pk.printf("%d %d\n", y, j)
+        for k in range(2):
+            pk.printf("print 2 %d\n", k)
+
+def main():
```

Comment:
can we format these as proper pytests so we can actually run them and have assertions in the end of each workunit?